genesiscommunitysuccess/blank-app-seed
# feat: enable initSSO configuration in genx FUI-899

[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

📓 &nbsp; **Related JIRA**

[FUI-899](https://genesisglobal.atlassian.net/browse/FUI-899)

🤔 &nbsp; **What does this PR do?**

Adds script that will be executed before the application is loaded.  This script has:

- Logic for fetching the list of providers
- The logic of checking the number of providers - if there is one, the script is executed, if there are more, we go to the application from which we can choose the provider
- Logic for checking whether we already have the necessary tokens to log in - if we do, we do not check the list of providers

Adds genx configuration to enable and disable a script/sso

🚀 &nbsp; **Where should the reviewer start?**

- client/web/public/initSSO.js
- .genx/prompts/api.js

📑 &nbsp; **How should this be tested?**

1. Check script
- You can manually override index.ejs - from blank-app-seed/.genx/plop/templates/index.ejs.hbs you can add script/onLoad function and delete blank-app tag
- In client/web/package.json in config add: ENABLE_SSO: true
- in initSSO.js set appHostUrl from which to fetch providers - by default it is location.host
- Run app - you should be redirected to SSO provider login

2. Genx
- Run genx
- Create new app
- Point it to Local Application seed and choose blank-app-seed on this branch
- When you get to the choice if you want to have a script, select yes
- After creating the application, check in your-app/client/web/public - whether you have index.ejs in which you have loaded the initSSO script and your-app/client/web/src/routes/config in which you have enabled sso 

:camera:  &nbsp; **Samples**

<img width="572" alt="Screenshot 2023-02-13 at 15 59 47" src="https://user-images.githubusercontent.com/22753085/218493049-f7cc90e4-7cab-4be4-9da9-e4dab2f0b79a.png">

1. Fetching providers
2. SSO login
3. Back to login page (is loaded because we have initSSO in our sessionStorage) and we are logged in
(In this example we are redirected to dev-position1 because I run it locally)

https://user-images.githubusercontent.com/22753085/218492321-1e7de580-2f0a-4098-aa1f-849f9a66b27a.mov


✅ &nbsp; **Checklist**
